media/Config/autoCompletePath.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const outputToInput = function(toolIndex, nextInputValue) {
+  for (let k = toolIndex + 1; k < oneToolList.length; k++) {
+    if (oneToolList[k].use === true) {
+      for (let l = 0; l < oneToolList[k].options.length; l++) {
+        if (oneToolList[k].options[l].optionName === 'input_path') {
+          oneToolList[k].options[l].optionValue = nextInputValue;
+          break;
+        }
+      }
+      break;
+    }
+  }
+};
+
+const makeOutputAuto = function(input, toolIndex, optionIndex) {
+  // because os maybe win32, divisor can be '\\'
+  let divisor = '/';
+  if (input.includes('\\')) {
+    divisor = '\\';
+  }
+  let paths = input.split(divisor);
+  let tmp = paths[paths.length - 1].split('.');
+  switch (oneToolList[toolIndex].type) {
+    case 'one-optimize': {
+      tmp.splice(tmp.length - 1, 0, 'opt');
+      paths[paths.length - 1] = tmp.join('.');
+      oneToolList[toolIndex].options[optionIndex].optionValue = paths.join(divisor);
+      break;
+    }
+    case 'one-quantize': {
+      if (tmp.includes('opt')) {
+        tmp[tmp.indexOf('opt')] = 'quantized';
+      } else {
+        tmp.splice(tmp.length - 1, 0, 'quantized');
+      }
+      paths[paths.length - 1] = tmp.join('.');
+      oneToolList[toolIndex].options[optionIndex].optionValue = paths.join(divisor);
+      break;
+    }
+    case 'one-pack': {
+      while (tmp.length > 1) {
+        tmp.splice(1, 1);
+      }
+      tmp[0] += '_pack';
+      paths[paths.length - 1] = tmp.join('.');
+      oneToolList[toolIndex].options[optionIndex].optionValue = paths.join(divisor);
+      break;
+    }
+    default: {
+      tmp[tmp.length - 1] = 'circle';
+      paths[paths.length - 1] = tmp.join('.');
+      oneToolList[toolIndex].options[optionIndex].optionValue = paths.join(divisor);
+    }
+  }
+  return oneToolList[toolIndex].options[optionIndex].optionValue;
+};
+
+// autoCompletePath make output_path based on input_path and copy former output_path to next
+// input_path for example, in import if input_path is 'filename.pb', then output_path will be
+// 'filename.circle' automatically and then in optimize output_path will be 'filename.opt.circle`
+// automatically
+const autoCompletePath = function(tool) {
```

Comment:
What I understand for now, a little bit, is that `autoCompletePath` will modify global single instance `oneToolList` array(?), which is declared in `tool.js`.
The codes here will find `input_path` and set `output_path` depending on the type
But the variable names are not `input` but `optionValue`

I'm not sure about this but, instead of setting directly like
```
oneToolList[toolIndex].options[optionIndex].optionValue = paths.join(divisor);
```
but use small functions like
```
input_file = get_input_file(...)
output_file = make_output_file(input_file, ...)
set_output_file(..., output_file)
```
and hide(?) direct accessing `oneToolList` with `i` and `j`  into those small functions...

If you agree on this, it would make code readability higher.
If you think this will take some time and that time is not much available, 
at lease please add some comments about what is `oneToolList[i]` is how it's attributes are accessed and interpreted.